version: "3"

services:
  db:
    image: library/postgres:12
    ports:
      - 5432:5432
    expose:
      - 5432
    environment:
      - POSTGRES_DB=comics
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - PGDATA=/tmp
    volumes:
      - ${POSTGRES_DATA_PATH}:/var/lib/postgresql/data:cached
      - tmp_data:/tmp
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 30s
      retries: 40

  comics:
    build: .
    depends_on:
      - db
    command: python manage.py runserver 0.0.0.0:8080
    restart: on-failure
    volumes:
      - ${COMICS_ROOT}:/comics
      - .:/code
    environment:
      - SECRET_KEY=${SECRET_KEY}
      - COMICS_ROOT=/comics
      - PYTHONBREAKPOINT=ipdb.set_trace
      - DATABASE_HOST=db
      - DATABASE_NAME=comics
      - DATABASE_USER=postgres
      - DATABASE_PASSWORD=postgres
      - DATABASE_PORT=5432
    ports:
      - "8080:8080"

volumes:
  tmp_data:
